### DISCUSSION TITLE
Tracking patch panels with splices with LC/SC and single strands

### DISCUSSION THREAD

**Discussion by bradbendy** [netbox-community/netbox#16941] [Q&A]
I've searched away and can't seem to find a answer on this, either I am missing something or it can't be done.

I have a patch panel with say 24 LC connectors on the front, the back of it 48 strands that would get spliced to either another patch panel then go to another set of 24 LC connectors or maybe 48 SC connectors for example. I don't see a way to track the front strand to match the rear strand since they are not 1:1. Real world use case is we are using LC connectors but BIDI optics so each physical port is now two cables which then are spliced on the back side.

Am I better off making a panel with 48 "LC" connectors and just track documentation wise that ports 1&2 are a real LC and then say 3&4 are BIDI. But then I have no way of connecting just one strand of that LC connector to a interface in netbox since can only connect the port to one interface. Maybe change the names around to "LC Port 1:1" and LC Port 1:2" and have to then modify them if im using it as a normal 2 strand LC maybe?

I looked at every example I could find the device types library and no luck. Curious as to what to others are doing in this world. Another example is we run LC panels at a POP for a PON network but we then breakout to single LC on the PON hardware but we want LC on the panels for the density reasons.  This is going to come up sooner or later with the DD optics that are taking two LC connectors as well (Juniper 800gb on PTX).

Thanks!

**Comment by candlerb** [comment]
Which Netbox version are you using?

In reasonably recent versions of Netbox, [v3.3.0](https://github.com/netbox-community/netbox/releases/v3.3.0) or later, for duplex interface connections you can make a cable which connects to one interface at one end and two frontports at the other. This documents that you are using two fibre strands, although it doesn't track exactly which one is transmit and which is receive.

> Am I better off making a panel with 48 "LC" connectors and just track documentation wise that ports 1&2 are a real LC and then say 3&4 are BIDI. But then I have no way of connecting just one strand of that LC connector to a interface in netbox since can only connect the port to one interface.

See above.

For this to work, you should model each duplex LC connector as being two frontports, i.e. 48 frontports in total. If your ODF labels a duplex LC connector as just port "1", then you could call them "1A" and "1B" in Netbox, or "1L" and "1R", or whatever makes sense to you.

This is not unreasonable: duplex LC connectors are genuinely two separate connectors with a plastic clip holding them together, just like duplex SC connectors.

For bidi interfaces you just connect the interface to one frontport, of course.

For versions of Netbox earlier than v3.3.0, you could connect to frontport 1A and then mark frontport 1B as "connected" without attaching a cable to it, which shows that it's in use.

**Reply by candlerb** [reply]
One other thing: often fibre infrastructure has crossed over pairs, like 1-2, 2-1, 3-4, 4-3 etc, and if so this will be important for your bidi connections.

You could make 48 rearports and write a script to hook them up with 48 individual cables correctly crossed over. Ouch.

But I'd be inclined to make two "models" of ODF: one which has F1,F2,F3,F4... connected to R:1,R:2,R:3,R:4..., and another where F1,F2,F3,F4... connect to R:2,R:1,R:4,R:3 etc. By using a different variant of ODF at each end, and a single cable between the two rearports, it *should* all come out correctly. (Untested by me, but I'd say bug report is valid if it doesn't work)

**Comment by bradbendy** [comment]
I'm on 3.6.1, I will have to play with this, but this sounds like what I need 100%. And I agree on the two "models" for ODF. I will make a test model up and give this a go, right now once I have cable connected I can only edit or delete it, so need to what I am doing wrong!
